src/components/Modals/NewFileModal.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from 'react';
+import { BsFillCheckCircleFill } from 'react-icons/bs';
+import toast from 'react-hot-toast';
+
+import { ConfirmModal } from './index';
+import examples from '../../examples';
+import { EditorService } from '../../services';
+import state from '../../state';
+
+interface TemplateListItemProps {
+  title: string;
+  description: React.FunctionComponent;
```

Comment:
I think that there should be better type to also "covers" ClassComponents.

```suggestion
  description: React.ComponentType;
```